aviary/interface/methods_for_level2.py

```diff
@@ -1394,13 +1533,22 @@ def add_linkage_constraint(self, phase_a, phase_b, var_b):
             self.traj.link_phases(phases[:7], vars=["TAS"], units='kn', ref=200.)
 
         elif self.mission_method is HEIGHT_ENERGY:
+
+            # connect mass and distance between all phases regardless of reserve / non-reserve status
             self.traj.link_phases(
                 phases, ["time", Dynamic.Mission.MASS, Dynamic.Mission.DISTANCE], connected=True)
 
+            # connect regular phases with each other if your are optimizing alt or mach
+            self._link_phases_helper_with_options(
+                self.regular_phases, 'optimize_altitude', Dynamic.Mission.ALTITUDE, ref=1.e4)
+            self._link_phases_helper_with_options(
+                self.regular_phases, 'optimize_mach', Dynamic.Mission.MACH)
+
+            # connect reserve phases with each other if your are optimizing alt or mach
             self._link_phases_helper_with_options(
```

Comment:
Should these be under an "If self.reserve_phases" so that they don't run otherwise?

Reply:
`_link_phases_helper_with_options()` checks if the phases dictionary that you pass it has 1 or less phases and doesn't make any connections if it does. We're only passing reserve_phases or regular_phases, to the function, so reserve and regular can't connect to each other by design.